Reorganize Resource Monitoring page

aimeeu opened the pull request:
Reorganize en/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md; put third-party tools in bullet lists.
Leave cAdvisor in its own paragraph because it is part of the kubelet binary.

Fixes #15711 
/assign @zacharysarah

aimeeu commented:
/assign @zacharysarah

tengqm commented on content/en/docs/tasks/debug-application-cluster/resource-usage-monitoring.md:
```diff
@@ -46,73 +47,61 @@ monitoring statistics by default:
-solutions. Sysdig is built on Open Source. [Sysdig and Sysdig Inspect](https://sysdig.com/opensource/inspect/) give you the 
-ability to freely perform troubleshooting, performance analyis and forensics. 
-
-### Google Cloud Monitoring
```
Looks to me this subsection has been drastically trimmed which is fine. However, we may want to remove the linked image resource if we are doing so.

aimeeu replied:
@tenqm Yes - the image has been marked for deletion (it's the second file changed, way at the bottom of the Files changed tab)

zacharysarah commented on content/en/docs/tasks/debug-application-cluster/resource-usage-monitoring.md:
```diff
@@ -46,73 +47,61 @@ monitoring statistics by default:
 
-The Kubelet acts as a bridge between the Kubernetes master and the nodes. It manages the pods and containers running on a machine. Kubelet translates each pod into its constituent containers and fetches individual container usage statistics from the container runtime, through the container runtime interface. For the legacy docker integration, it fetches this information from cAdvisor.  It then ex ...
+The Kubelet acts as a bridge between the Kubernetes master and the nodes. 
+It manages the pods and containers running on a machine. Kubelet translates each pod 
```
s/Kubelet/The kubelet

See https://github.com/kubernetes/website/pull/14831#issuecomment-518871961 for context.

aimeeu replied:
updated

zacharysarah commented on content/en/docs/tasks/debug-application-cluster/resource-usage-monitoring.md:
```diff
@@ -46,73 +47,61 @@ monitoring statistics by default:
 ### cAdvisor
 
-cAdvisor is an open source container resource usage and performance analysis agent. It is purpose-built for containers and supports Docker containers natively. In Kubernetes, cAdvisor is integrated into the Kubelet binary. cAdvisor auto-discovers all containers in the machine and collects CPU, memory, filesystem, and network usage statistics. cAdvisor also provides the overall machine usage by an ...
+cAdvisor is an open source container resource usage and performance analysis 
```
Per #15576, this section no longer meets content guidelines because it's:
* not in kubernetes or kubernetes-sigs
* not a CNCF project

So L59-69 need to be removed.

aimeeu replied:
removed

zacharysarah commented on content/en/docs/tasks/debug-application-cluster/resource-usage-monitoring.md:
```diff
@@ -46,73 +47,61 @@ monitoring statistics by default:
-
-
-{{< figure src="/images/docs/gcm.png" alt="Google Cloud Monitoring dashboard example" title="Google Cloud Monitoring dashboard example" caption="This dashboard shows cluster-wide resource usage." >}}
+* [Elasticsearch and Kibana (Elastic Stack)](https://www.elastic.co/products/elastic-stack) The Elastic stack has a robust feature set for
```
Per #15576, this content no longer meets requirements. Remove L73-74.

aimeeu replied:
removed

zacharysarah commented on content/en/docs/tasks/debug-application-cluster/resource-usage-monitoring.md:
```diff
@@ -46,73 +47,61 @@ monitoring statistics by default:
+* [Elasticsearch and Kibana (Elastic Stack)](https://www.elastic.co/products/elastic-stack) The Elastic stack has a robust feature set for
+ logging, reporting, and visualizing log data.
+
+* [Google Stackdriver](https://cloud.google.com/stackdriver/) is a hosted monitoring service you can use to
```
Per #15576, this content no longer meets standards. Remove L76-77.

zacharysarah commented on content/en/docs/tasks/debug-application-cluster/resource-usage-monitoring.md:
```diff
@@ -46,73 +47,61 @@ monitoring statistics by default:
+* [Google Stackdriver](https://cloud.google.com/stackdriver/) is a hosted monitoring service you can use to
+visualize and alert on important metrics in your application. 
+
+* [Prometheus](https://prometheus.io) can natively monitor kubernetes, nodes, and 
```
Per #15576, this content meets standards (Prometheus is a CNCF project) and can remain. 👍

zacharysarah commented on content/en/docs/tasks/debug-application-cluster/resource-usage-monitoring.md:
```diff
@@ -46,73 +47,61 @@ monitoring statistics by default:
+  visualizing your data. Prometheus is also a supported data source for 
+  [Grafana](https://prometheus.io/docs/visualization/grafana/).
+
+* [Sysdig](http://sysdig.com) provides full spectrum container and platform 
```
Per #15576, this content no longer meets minimum standards. Remove L89-96.

aimeeu replied:
removed

zacharysarah commented on content/en/docs/tasks/debug-application-cluster/resource-usage-monitoring.md:
```diff
@@ -46,73 +47,61 @@ monitoring statistics by default:
-### New Relic Kubernetes monitoring integration
-
-[New Relic Kubernetes](https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/kubernetes-monitoring-integration) integration provides increased visibility into the performance of your Kubernetes environment. New Relic's Kubernetes integration instruments the container orchestration layer by reporting metrics from Kubernetes objects. The integration gives you insight  ...
+* With the [Kubernetes Job Monitor](https://github.com/pietervogelaar/kubernetes-job-monitor) dashboard, a Cluster Administrator can see which
```
Per #15576, this content no longer meets minimum standards. Remove L100-101.

aimeeu replied:
removed

zacharysarah commented on content/en/docs/tasks/debug-application-cluster/resource-usage-monitoring.md:
```diff
@@ -46,73 +47,61 @@ monitoring statistics by default:
-Create your own custom queries and charts in Insights from automatically reported data.
-Create alert conditions on Kubernetes data.
-Learn more on this [page](https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/kubernetes-monitoring-integration).
+* [New Relic Kubernetes](https://newrelic.com/platform/kubernetes) monitors
```
Per #15576, this content no longer meets minimum standards. Remove L103-105.

aimeeu replied:
removed

thecrudge commented:
/lgtm once changes implemented

aimeeu commented:
@zacharysarah I removed third-party references and reworked the page -- removed bullet points, consolidated information into paragraphs. Let me know if this works. Reread the style guide and removed the single back-ticks around metrics-server because it's a project not a command - right?

xiangpengzhao commented:
This LGTM. Defer to @zacharysarah for final approval.
/lgtm

aimeeu commented:
/close

zacharysarah commented on content/en/docs/tasks/debug-application-cluster/resource-usage-monitoring.md:
```diff
@@ -29,90 +29,49 @@ monitoring statistics by default:
   lightweight short-term in-memory store.
   
-- A [**full metrics pipeline**](#full-metrics-pipelines), such as Prometheus, gives you access to richer
+- A [**full metrics pipeline**](#full-metrics-pipeline), such as Prometheus, gives you access to richer
```
Good catch. 👍
